osu.Game/Modes/Replays/FramedReplay.cs

```diff
@@ -3,67 +3,39 @@
 
 using System;
 using System.Collections.Generic;
-using System.IO;
 using osu.Framework.Extensions.IEnumerableExtensions;
 using osu.Framework.Input;
 using osu.Framework.MathUtils;
 using osu.Game.Input.Handlers;
-using osu.Game.IO.Legacy;
 using OpenTK;
 using OpenTK.Input;
 using KeyboardState = osu.Framework.Input.KeyboardState;
 using MouseState = osu.Framework.Input.MouseState;
 
-namespace osu.Game.Modes
+namespace osu.Game.Modes.Replays
 {
-    public class LegacyReplay : Replay
+    public abstract class FramedReplay : Replay

```

Comment:
There might be one level of abstraction too much here. Consider that Taiko will need to derive `FramedReplayInputHandler` to override `GetPendingStates`, however it still needs to be able to be able to load from legacy replays which isn't easily possible due to `LegacyFramedReplay : FramedReplay`